### DISCUSSION TITLE
[Feature Request]: Easy S2S (Site to Site) configuration

### DISCUSSION THREAD

**Discussion by the-hotmann** [wg-easy/wg-easy#1054] [Ideas]
`wg-easy` makes it very easy to connect clients to the wireguard server by clicking `Add client` button, but does not very directly show, how to add other wireguard (`wg-easy`) Servers, so that you as easy as adding clients, can add site-to-site Wireguard connections.

I think that would be awesome. Maybe adding another button: `Add Server`, would be the perfect solution, which then lets you choose the settings for the tunnle.

I am open for discussion about this.

**Comment by the-hotmann** [comment]
I'd like to incorporate the following points:

1. Acknowledging that WireGuard operates on a peer-to-peer basis, it doesn't distinguish between "Server" and "Client," only peers.
2. The significant distinction between peers acting as "clients" and those as "servers" lies in the mapping of IPs within the tunnel, usually creating either a 1:n (for clients) or n:n relationship (for site-to-site).

However, the existing implementation lacks differentiation between devices/clients and servers/site-to-site connections. It would be beneficial to manage them in separate lists, visually separated. Maybe there should be a "client" page and a "server/site-to-site" page.

**Comment by anktababa** [comment]
Hello, 
I was looking for creating a site to site configuration (In fact this is more hub and spoke from my POV, as two sites can communicate together using the hub which is a the wireguard server)

To achieve this i guess we need :
- Setup the wg server using wg-easy
- Create two clients
- More "complex part" is : Allow the communication from Site A to Site B and vice versa (Example : Site A : 192.168.**0**.X/24 , Site B : 192.168.**1**.X/24 ) This should be done using the allowed IP parameters i guess
- Add the correct routes on the WG-Server

Thanks

**Comment by tokKurumi** [comment]
Any updates / thoughts?
